### PR TITLE
Cal 7

### PR DESCRIPTION
**Pull request by xorver**

### CODE REVIEW

File: src/main/java/pl/edu/agh/student/config/RedisCloudSessionConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.Protocol;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+@Configuration
+@PropertySource("classpath:config.properties")
+@Profile("default")
+public class RedisCloudSessionConfiguration {
```
Comment (sylgas) [inline src/main/java/pl/edu/agh/student/config/RedisCloudSessionConfiguration.java]: AppConfig, MongoConfig, RedisCloudSessionConfig?

File: src/main/java/pl/edu/agh/student/config/RedisCloudSessionConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.Protocol;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+@Configuration
+@PropertySource("classpath:config.properties")
```
Comment (sylgas) [inline src/main/java/pl/edu/agh/student/config/RedisCloudSessionConfiguration.java]: You haven't add/read any property to/from this file, is this needed?

File: src/main/java/pl/edu/agh/student/config/RedisDevSessionConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import redis.clients.jedis.Protocol;
+import redis.embedded.RedisServer;
+
+@Configuration
+@Profile("dev")
+public class RedisDevSessionConfiguration {
```
Comment (sylgas) [inline src/main/java/pl/edu/agh/student/config/RedisDevSessionConfiguration.java]: RedisDevSessionConfig?

File: README.md [hunk 4]
```diff
@@ -41,3 +41,19 @@ W celu automatycznej kompilacji plików .coffee należy wymagane jest skonfiguro
...
+z bazy Redis. Informacje połączenia pobierane są ze zmiennej środowiskowej
+exportowanej przez heroku: REDISTOGO_URL. 
+
+Na potrzeby developmentu i szybkiego uruchamiania aplikacji na jednym węźle,
+można skorzystać z wbudowanej bazy Redis, która aktywuje się po włączeniu profilu
+springowego "dev". Można to zrobić np. poprzez dodanie:
+
+```
+   Intellij / Run / Edit Configurations... / Tomcat Server / <config_name> / Startup/Connection / Environment Variables / Add
+   spring.profiles.active=dev
```
Comment (sylgas) [inline README.md:58]: still getting:
 org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'enableRedisKeyspaceNotificationsInitializer' defined in class path resource [org/springframework/session/data/redis/config/annotation/web/http/RedisHttpSessionConfiguration.class]: Invocation of init method failed; nested exception is org.springframework.data.redis.RedisConnectionFailureException: Cannot get Jedis connection; nested exception is redis.clients.jedis.exceptions.JedisConnectionException: Could not get a resource from the pool

File: src/main/java/pl/edu/agh/student/config/RedisCloudSessionConfiguration.java (hunk 1 above)
Comment (xorver) [inline src/main/java/pl/edu/agh/student/config/RedisCloudSessionConfiguration.java]: done

File: src/main/java/pl/edu/agh/student/config/RedisCloudSessionConfiguration.java (hunk 2 above)
Comment (xorver) [inline src/main/java/pl/edu/agh/student/config/RedisCloudSessionConfiguration.java]: done

File: src/main/java/pl/edu/agh/student/config/RedisDevSessionConfiguration.java (hunk 3 above)
Comment (xorver) [inline src/main/java/pl/edu/agh/student/config/RedisDevSessionConfiguration.java]: done

File: README.md (hunk 4 above)
Comment (xorver) [inline README.md:58]: Do you run it in "dev" profile? Try swapping DevConfig and CloudConfig spring profiles (default/dev) so we will know if run configuration is the cause of problem.

Comment (sylgas) [inline README.md:58]: after swapping error still occurs

**Comment by xorver** [comment]
It's already merged

**Pull request closed without merging**